From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Fri, 5 Jun 2026 17:15:18 +0800
Subject: local-pthread_once

Note: Keep the 2.41 symbol until we rebuild packages against the 2.42

symbol.
---
 htl/Versions                          | 1 +
 sysdeps/htl/pt-once.c                 | 3 +++
 sysdeps/mach/hurd/i386/libc.abilist   | 1 +
 sysdeps/mach/hurd/x86_64/libc.abilist | 1 +
 4 files changed, 6 insertions(+)

diff --git a/htl/Versions b/htl/Versions
index cac34e6..95f4fc9 100644
--- a/htl/Versions
+++ b/htl/Versions
@@ -201,6 +201,7 @@ libc {
     pthread_mutexattr_setpshared;
     pthread_mutexattr_setrobust; pthread_mutexattr_setrobust_np;
     pthread_mutexattr_settype;
+    pthread_once;
     pthread_sigmask;
   }
 
diff --git a/sysdeps/htl/pt-once.c b/sysdeps/htl/pt-once.c
index 5737033..447a486 100644
--- a/sysdeps/htl/pt-once.c
+++ b/sysdeps/htl/pt-once.c
@@ -57,6 +57,9 @@ __pthread_once (pthread_once_t *once_control, void (*init_routine) (void))
 libc_hidden_def (__pthread_once)
 versioned_symbol (libc, __pthread_once, pthread_once, GLIBC_2_42);
 
+#if OTHER_SHLIB_COMPAT (libpthread, GLIBC_2_12, GLIBC_2_42)
+compat_symbol (libpthread, __pthread_once, pthread_once, GLIBC_2_41);
+#endif
 #if OTHER_SHLIB_COMPAT (libpthread, GLIBC_2_12, GLIBC_2_42)
 compat_symbol (libpthread, __pthread_once, pthread_once, GLIBC_2_12);
 #endif
diff --git a/sysdeps/mach/hurd/i386/libc.abilist b/sysdeps/mach/hurd/i386/libc.abilist
index 08dcd85..cf9c963 100644
--- a/sysdeps/mach/hurd/i386/libc.abilist
+++ b/sysdeps/mach/hurd/i386/libc.abilist
@@ -2653,6 +2653,7 @@ GLIBC_2.41 pthread_mutexattr_setpshared F
 GLIBC_2.41 pthread_mutexattr_setrobust F
 GLIBC_2.41 pthread_mutexattr_setrobust_np F
 GLIBC_2.41 pthread_mutexattr_settype F
+GLIBC_2.41 pthread_once F
 GLIBC_2.41 pthread_sigmask F
 GLIBC_2.42 __inet_ntop_chk F
 GLIBC_2.42 __inet_pton_chk F
diff --git a/sysdeps/mach/hurd/x86_64/libc.abilist b/sysdeps/mach/hurd/x86_64/libc.abilist
index c17d539..471331e 100644
--- a/sysdeps/mach/hurd/x86_64/libc.abilist
+++ b/sysdeps/mach/hurd/x86_64/libc.abilist
@@ -2330,6 +2330,7 @@ GLIBC_2.41 pthread_mutexattr_setpshared F
 GLIBC_2.41 pthread_mutexattr_setrobust F
 GLIBC_2.41 pthread_mutexattr_setrobust_np F
 GLIBC_2.41 pthread_mutexattr_settype F
+GLIBC_2.41 pthread_once F
 GLIBC_2.41 pthread_sigmask F
 GLIBC_2.42 __inet_ntop_chk F
 GLIBC_2.42 __inet_pton_chk F
